### DISCUSSION TITLE
How do you preserve the real Client IP with kube-proxy and Traefik?

### DISCUSSION THREAD

**Discussion by marylieh** [cilium/cilium#35678] [Q&A] [answered]
Hello, 
i have a Problem where I can't see the the real client IPs in my Traefik access log. I only see the cilium interface IP of the Node where Traefik is running. 
The Traefik service is from the type LoadBalancer and I use ciliums integrated Layer 4 LoadBalancer to allocate an IP Address.
After much googling I found out that I have to either use `externalTrafficPolicy: local` or use ciliums kube-proxy replacement to enable DSR.
I would use PROXY Protocol or correct X-forwarded-for headers when I would know how to configure these in cilium. 

So my question ist. Is there any way I can use the Cilium LoadBalancer and preserve the real client IP without the hassle of switching to cilium kube-proxy replacement or use externalTrafficPolicy? This is pretty important to me because otherwise it is impossible to block access from specific IPs. 

Best regards
Marie

**Comment by joestringer** [comment, marked as answer]
If you are not running kube-proxy replacement, then the component that causes the loss of original client IP is kube-proxy. Therefore you would need to ask the kube-proxy community about mitigations. In general, the Cilium solution to this problem is documented [here](https://docs.cilium.io/en/latest/network/kubernetes/kubeproxy-free/#client-source-ip-preservation).
